protocol/bin-prot/src/value/ser.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::Value;
+
+use serde::ser::{SerializeStruct, SerializeTuple};
+use serde::Serialize;
+
+impl Serialize for Value {
+    #[inline]
```

Comment:
@hanabi1224 would it be simple to benchmark this?

Reply:
TBH I don't think it makes any difference. It will very likely be inlined w/o this attribute with the below cargo manifest config. The attribute is only a hint and may be ignored.

> The cold and inline attributes give suggestions to generate code in a way that may be faster than what it would do without the hint. The attributes are only hints, and may be ignored.

> #[inline] suggests performing an inline expansion. 
#[inline(always)] suggests that an inline expansion should always be performed.
#[inline(never)] suggests that an inline expansion should never be performed.

```toml
[profile.release]
codegen-units = 1
lto = true
opt-level = 3
overflow-checks = false
```